libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -1058,6 +1063,18 @@ void AP_Logger_File::_io_timer(void)
         ::fsync(_write_fd);
         last_io_operation = "";
 #endif
+
+        if (_need_rtc_update) {
+            uint64_t utc_usec;
+            if (AP::rtc().get_utc_usec(utc_usec)) {
+                struct utimbuf t;
```

Comment:
Suggest:

```
const struct utimbuf t{
    modtime: utc_usec / (1000UL * 1000UL),
    actime: t.modtime
};
```

As things currently stand it's not clear to the reader if this struct is fully initialised or contains garbage from the stack.

Reply:
.... OK, suggestion is not quite right here...

Reply:
fixed to be only ChibiOS